### ISSUE TITLE
native getShopPedQueryOutfit always return 0

### ISSUE THREAD

**Issue by Hanashi** [altmp/altv-issues#944]
**Client/server version**

`Build 2.23, branch release`

**Current behavior**
I tried to execute the native `getShopPedQueryOutfit`. The output is always `0`. (look at steps to reproduce)

**Expected behavior**
The output should be an object ([look at this example](https://gist.github.com/root-cause/e77c1b1e003402687291ea76b0187677#file-structs-h-L1)).

**Steps to reproduce**

Try the following code:
```TypeScript
let outfits = native.setupShopPedOutfitQuery(3, false);
console.log('Outfits: ' + outfits);

for (let i = 0; i < outfits; i++) {
    let [_, outfit] = native.getShopPedQueryOutfit(i);
    console.log(outfit);
}
```

**Context (environment)**

I want to get alle outfits for the multiplayer characters.

**Comment by LeonMrBonnie** [comment]
This might be an error in the client JS module handling of struct return values, so someone should investigate into that further.

**Comment by LeonMrBonnie** [comment]
This is an error in alt:V core, the return type of that function is marked as int32, not struct.

**Comment by FabianTerhorst** [comment]
This needs to return a memory buffer so it can be read. Same handling as with head blend data getter.

**Comment by LeonMrBonnie** [comment]
> 
> 
> This needs to return a memory buffer so it can be read. Same handling as with head blend data getter.

Right, but the native return type for the native posted is marked as `alt::INative::Type::ARG_INT32` and not `alt::INative::Type::ARG_STRUCT`

**Comment by FabianTerhorst** [comment]
the native doesnt has to return a struct. The memory buffer works with that native definition. Because the memory buffer is giving in as a pointer via a parameter.
